client/pcr_test.go

```diff
@@ -125,3 +262,71 @@ func TestCheckContainedPCRs(t *testing.T) {
 		t.Fatalf("empty pcrs is always validate")
 	}
 }
+
+func TestCheckContainedPCRsDirect(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	sel := client.FullPcrSelDirect(tpm.AlgSHA256)
+	baseline, err := client.ReadPCRsDirect(thetpm, sel)
+	if err != nil {
+		t.Fatalf("Failed to Read PCRs: %v", err)
+	}
+
+	pcrs, err := CreatePCRSelection([]int{test.DebugPCR})
+	if err != nil {
+		t.Fatalf("Failed to create PCRSelection")
+	}
+	toBeCertified, err := client.ReadPCRsDirect(thetpm, tpms.PCRSelection{Hash: tpm.AlgSHA256, PCRSelect: pcrs})
+	if err != nil {
+		t.Fatalf("failed to read pcrs %v", err)
+	}
+	if err := internal.CheckSubset(toBeCertified, baseline); err != nil {
+		t.Fatalf("Validation should pass: %v", err)
+	}
+
+	pcrExtend := tpm2direct.PCRExtend{
+		PCRHandle: tpm2direct.AuthHandle{
+			Handle: tpm.Handle(test.DebugPCR),
+			Auth:   tpm2direct.PasswordAuth(nil),
+		},
+		Digests: tpml.DigestValues{
+			Digests: []tpmt.HA{
+				{
+					HashAlg: tpm.AlgSHA256,
+					Digest:  bytes.Repeat([]byte{0x00}, sha256.Size),
+				},
+			},
+		},
+	}
+	if err = pcrExtend.Execute(thetpm); err != nil {
+		t.Fatalf("failed to extend pcr for test %v", err)
+	}
+
+	pcrs, err = CreatePCRSelection([]int{1, 3, test.DebugPCR})
+	if err != nil {
+		t.Fatalf("Failed to create PCRSelection")
+	}
+	toBeCertified, err = client.ReadPCRsDirect(thetpm, tpms.PCRSelection{Hash: tpm.AlgSHA256, PCRSelect: pcrs})
+	if err != nil {
+		t.Fatalf("failed to read pcrs %v", err)
+	}
+	if err := internal.CheckSubset(toBeCertified, baseline); err == nil {
+		t.Fatalf("Validation should pass: %v", err)
```

Comment:
bad copy paste? Should the error message say something like above:
```
	if err := internal.CheckSubset(toBeCertified, baseline); err == nil {
		t.Fatalf("validation should fail due to PCR %d changed", test.DebugPCR)
	}
```

Reply:
Yup, fixed! Resolved.